### PR TITLE
GUI Nav Tests

### PR DESCRIPTION
**Pull request by ejkorol**
### GUI Nav Tests

_Adds tests for the nav components_

Adds tests and snapshots for:
- `@/layout.tsx`
- `@/components/navigation/header.tsx`
- `@/components/navigation/footer.tsx`
- `@/components/navigation/sidebar.tsx`

### CODE REVIEW

File: src/gui/test/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}))
+vi.mock('@/app/dashboard.jsx', () => ({
+  Dashboard: () => <div>Mock Dashboard</div>,
+}))
+vi.mock('@/app/explorer.jsx', () => ({
+  Explorer: () => <div>Mock Explorer</div>,
+}))
+vi.mock('@/app/error-found.jsx', () => ({
+  ErrorFound: () => <div>Mock Error Found</div>,
+}))
```
Comment (ruyadorno) [inline src/gui/test/layout.tsx:28]: same as `src/gui/test/app/layout.tsx`

File: src/gui/test/components/navigation/sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+vi.spyOn(ThemeProviderModule, 'useTheme').mockImplementation(() => ({
+  theme: 'light',
+  setTheme: setThemeMock,
+}))
+vi.mock('useGraphStore', () => ({
+  useGraphStore: vi.fn(),
+}))
+vi.mock('useTheme', () => ({
+  useTheme: vi.fn(),
+}))
```
Comment (ruyadorno) [inline src/gui/test/components/navigation/sidebar.tsx:26]: I'm curious about `vi.fn()` what is it doing in this test?

**Review by ruyadorno — CHANGES_REQUESTED** [review]

File: src/gui/test/app/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import { afterEach, test, vi, expect } from 'vitest'
+import { render, cleanup } from '@testing-library/react'
+import html from 'diffable-html'
+import { useGraphStore as useStore } from '@/state/index.js'
+import Layout from '@/layout.jsx'
+
+vi.mock('useGraphStore', () => ({
+  useGraphStore: vi.fn(),
+}))
+
```
Comment (ruyadorno) [inline src/gui/test/app/layout.tsx:10]: You shouldn't need to mock `useGraphStore`, you can refer to [selected-items tests](https://github.com/vltpkg/vltpkg/blob/main/src/gui/test/components/explorer-grid/selected-item.tsx) for a reference of how to test using the zustand store.

File: src/gui/test/app/__snapshots__/layout.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    aria-label="Notifications (F8)"
+    tabindex="-1"
+    style="pointer-events: none;"
+  >
+    <ol
+      tabindex="-1"
+      class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
+    >
+    </ol>
+  </div>
```
Comment (ruyadorno) [inline src/gui/test/app/__snapshots__/layout.tsx.snap:91]: I see that this is repeated in many snapshots, is this intentional?

Comment (ejkorol) [reply src/gui/test/app/__snapshots__/layout.tsx.snap]: Looks like this is the `<Toaster />`'s `ToastViewport` component from `components/ui/toast.tsx`. I've added the mock for it in the `layout.tsx` test!

Comment (ruyadorno) [reply src/gui/test/app/__snapshots__/layout.tsx.snap]: ooohhh I see 👍 makes sense!

File: src/gui/test/components/navigation/footer.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import html from 'diffable-html'
+import { useGraphStore as useStore } from '@/state/index.js'
+import { Footer } from '@/components/navigation/footer.jsx'
+
+vi.mock('useTheme', () => ({
+  useTheme: vi.fn(),
+}))
+vi.mock('useGraphStore', () => ({
+  useGraphStore: vi.fn(),
+}))
```
Comment (ruyadorno) [inline src/gui/test/components/navigation/footer.tsx:12]: same as before, I don't think we need / should be mocking these, is `useTheme` even still a thing?

File: src/gui/test/components/navigation/header.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { test, expect, afterEach, vi } from 'vitest'
+import { render, cleanup } from '@testing-library/react'
+import html from 'diffable-html'
+import { useGraphStore as useStore } from '@/state/index.js'
+import { Header } from '@/components/navigation/header.jsx'
+
+vi.mock('useGraphStore', () => ({
+  useGraphStore: vi.fn(),
+}))
```
Comment (ruyadorno) [inline src/gui/test/components/navigation/header.tsx:9]: same

**Review by ruyadorno — APPROVED** [review]
LGTM

**Pull request merged**